### PR TITLE
Strip vendored dSYMs during embed script phase

### LINKED ISSUE #7111: CocoaPods leaves dSYMs for unused architectures in archive
## What did you do?
1. Created a new empty iOS project with Xcode 9.0. Then closed it.
2. Ran `pod init`.
3. In the pod file, I added one framework `TealiumIOS` ([docs here](https://github.com/tealium/tealium-ios))
4. Ran `pod install` to bring in TealiumIOS.
5. Opened the `SampleApp.xcworkspace` with Xcode and archived my application.
6. Next, I inspected the symbols for each architecture in the archive's `dSYMs` directory. Here they are after `cd`-ing to my archive and running `dwarfdump`:

```
dSYMs $ cd ~/Library/Developer/Xcode/Archives/2017-10-06/SampleApp\ 10-6-17\,\ 3.21\ PM.xcarchive/dSYMs
dSYMs $ dwarfdump --uuid *
UUID: C56F7ABF-DFD7-3ABC-A9F6-0B172E7B9F10 (arm64) SampleApp.app.dSYM/Contents/Resources/DWARF/SampleApp
UUID: EA990B6D-CAF9-3B87-8D99-D04A4FA36537 (i386) TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS
UUID: 1306BCB7-953D-3FC5-956E-40B85B8986A1 (x86_64) TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS
UUID: CF845CCB-4E31-379F-8835-DA3B16734722 (armv7) TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS
UUID: B353F154-B6B4-3FCA-AA18-C18E400EC139 (arm64) TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS
```

## What did you expect to happen?

I would expect that in addition to stripping the invalid architectures here, that CocoaPods would also strip the invalid symbol files (if any):
https://github.com/CocoaPods/CocoaPods/blob/2fb2db21016f82806d3940013bda30ffa941151b/lib/cocoapods/generator/embed_frameworks_script.rb#L123-L138 

More precisely I would expect that the symbol files left in my archive with UUIDs `EA990B6D-CAF9-3B87-8D99-D04A4FA36537` (`i386`) and `1306BCB7-953D-3FC5-956E-40B85B8986A1` (`x86_64`) would be stripped from my archive because the App Store does not need or expect symbols for these architectures. In fact, iTunes Connect now warns about uploads of symbols that don't correspond to any binaries:
```
Too many symbol files - These symbols have no corresponding slice in any binary [1306BCB7-953D-3FC5-956E-40B85B8986A1.symbols, EA990B6D-CAF9-3B87-8D99-D04A4FA36537.symbols]
```

## What happened instead?

The `i386` and `x86_64` symbols were left in the archive.

### Stack

```
   CocoaPods : 1.3.1
        Ruby : ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin16]
    RubyGems : 2.5.1
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 9.0 (9A235)
         Git : git version 2.13.5 (Apple Git-94)
Ruby lib dir : /Users/aconnolly/.rvm/rubies/ruby-2.3.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 5a6cc2e32ae83d327d318296c4fb69125e8fc3ab
```

### Installation Source

```
Executable Path: /Users/aconnolly/.rvm/gems/ruby-2.3.0/bin/pod
```

### Plugins

```
claide-plugins        : 0.9.2
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
platform :ios, '10.0'

target 'SampleApp' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for SampleApp
  pod 'TealiumIOS', '5.3.0'

  target 'SampleAppTests' do
    inherit! :search_paths
    # Pods for testing
  end

  target 'SampleAppUITests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```
## Project that demonstrates the issue
https://github.com/chefnobody/SampleApp

## Thoughts / Findings:
- It seems Carthage had a similar problem back in 2016 and resolved it (I think): https://github.com/Carthage/Carthage/pull/1023
- This issue has been identified elsewhere by other Framework makers: https://github.com/mapbox/mapbox-gl-native/issues/10137
- I think this is the step in the generated `-frameworks.sh` where the copy of all dSYMs happens. But I could be wrong: https://github.com/CocoaPods/CocoaPods/blob/2fb2db21016f82806d3940013bda30ffa941151b/lib/cocoapods/generator/embed_frameworks_script.rb#L98-L105

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #7111

### CODE REVIEW

**Comment by dnkoutso** [comment]
Integration specs PR https://github.com/CocoaPods/cocoapods-integration-specs/pull/126

**Comment by dnkoutso** [comment]
Should go green.

**Comment by dnkoutso** [comment]
From the archive folder:

```
dnkoutso@dnkoutso ~/Development/ios $ lipo -info /Users/dnkoutso/Library/Developer/Xcode/Archives/2017-10-06/SampleApp\ 10-6-17\,\ 3.41\ PM.xcarchive/dSYMs/TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS 
Architectures in the fat file: /Users/dnkoutso/Library/Developer/Xcode/Archives/2017-10-06/SampleApp 10-6-17, 3.41 PM.xcarchive/dSYMs/TealiumIOS.framework.dSYM/Contents/Resources/DWARF/TealiumIOS are: arm64 
```

File: lib/cocoapods/generator/embed_frameworks_script.rb [hunk 1]
```diff
@@ -102,6 +102,15 @@ def script
               echo "rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --filter \\"- CVS/\\" --filter \\"- .svn/\\" --filter \\"- .git/\\" --filter \\"- .hg/\\" --filter \\"- Headers\\" --filter \\"- PrivateHeaders\\" --filter \\"- Modules\\" \\"${source}\\" \\"${DWARF_DSYM_FOLDER_PATH}\\""
               rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${source}" "${DWARF_DSYM_FOLDER_PATH}"
             fi
+
+            local basename
+            basename="$(basename -s .framework.dSYM "$source")"
+            binary="${DWARF_DSYM_FOLDER_PATH}/${basename}.framework.dSYM/Contents/Resources/DWARF/${basename}"
+
+            # Strip invalid architectures so "fat" simulator / device frameworks work on device
```
Comment (paulb777) [inline lib/cocoapods/generator/embed_frameworks_script.rb]: Should the comment say "Strip invalid dSYMs" instead of "architectures"?

Comment (dnkoutso) [reply lib/cocoapods/generator/embed_frameworks_script.rb]: I think it's fine, it is actually stripping architectures :)

**Pull request merged**